tutorials/tools/Text_Denormalization_(Inverse_Text_Normalization).ipynb

```diff
@@ -0,0 +1,368 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "heavy-distance",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print([byte for byte in bytes('fst', 'utf-8')])\n",
+    "\n",
+    "# create an acceptor from a string\n",
+    "pynini.accep('fst')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "brave-avatar",
+   "metadata": {},
+   "source": [
+    "Here `0` - is a start note, `1` and `2` are the accept nodes, while `3` is a finite state.\n",
+    "By default (token_type=\"byte\", `Pynini` interprets the string as a sequence of bytes, assigning one byte per arc. \n",
+    "\n",
+    "A finite state transducer (FST) not only matches the pattern but also produces output according to the defined transitions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "paperback-female",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create an FST\n",
+    "pynini.cross('fst', 'FST')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "blond-hypothetical",
+   "metadata": {},
+   "source": [
+    "Pynini supports the following operations:\n",
+    "\n",
+    "- `closure` - Computes concatenative closure.\n",
+    "- `compose` - Constructively composes two FSTs.\n",
+    "- `concat` - Computes the concatenation (product) of two FSTs.\n",
+    "- `difference` - Constructively computes the difference of two FSTs.\n",
+    "- `invert`  - Inverts the FST's transduction.\n",
+    "- `optimize` - Performs a generic optimization of the FST.\n",
+    "- `project` - Converts the FST to an acceptor using input or output labels.\n",
+    "- `shortestpath` - Construct an FST containing the shortest path(s) in the input FST.\n",
+    "- `union`- Computes the union (sum) of two or more FSTs.\n",
+    "\n",
+    "\n",
+    "The list of most commonly used `Pynini` operations could be found [https://github.com/kylebgorman/pynini/blob/master/CHEATSHEET](https://github.com/kylebgorman/pynini/blob/master/CHEATSHEET). \n",
+    "\n",
+    "Pynini examples could be found at [https://github.com/kylebgorman/pynini/tree/master/pynini/examples](https://github.com/kylebgorman/pynini/tree/master/pynini/examples).\n",
+    "Use `help()` to explore the functionality. For example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "arctic-firewall",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "help(pynini.union)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "continued-optimum",
+   "metadata": {},
+   "source": [
+    "# NeMo ITN API"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "twenty-terrorist",
+   "metadata": {},
+   "source": [
+    "NeMo ITN defines the following APIs that are called in sequence:\n",
+    "\n",
+    "- `classify()` - creates a linear automaton from the input string and composes it with the final classification WFST, which transduces numbers and inserts semantic tags.  \n",
```

Comment:
they have different names now, could you use signature in current ode?
classify i think is find_tag + select_tag,
verbalize is find_veerbalize + select_verbalize
etc